src/service.rs

```diff
@@ -1221,7 +1230,22 @@ impl Service {
                     }
                 }
             } else {
-                return false; // Didn't pass the table filter remove the peer
+                // Is either non-contactable or didn't pass the table filter. If it exists in the
+                // routing table, remove it.
+                match self.kbuckets.write().entry(&key) {
+                    kbucket::Entry::Present(entry, _) if entry.value().seq() < enr.seq() => {
+                        entry.remove()
+                    }
+                    kbucket::Entry::Pending(mut entry, _) => {
+                        if entry.value().seq() < enr.seq() {
+                            entry.remove()
+                        }
+                    }
+                    _ => {}
+                }
```

Comment:
nit:
is there be a situation for legit storing a non contactable ENR?
If not can we (instead) log an error and have a `debug_unreachable` if it exists in `kbuckets` like in
https://github.com/sigp/discv5/blob/6707bcf3da350ff273208bb38dc29acc4cc4850c/src/service.rs#L598

Reply:
The entry here is about finding a new ENR. 

So we could have an ENR that is contactable and we store it in the routing table. Then later, we find they have changed their ENR to something non-contactable and we see an updated version. Because the key in the table is the node-id, it can be that the earlier version was contactable but the new one is not. So we remove the old one here and don't store the new one. 

I can't wrap this in a debug only mode, because this is a valid case. 

For the sake of speed, I"ll merge this PR and if my analysis here is wrong we can make a future PR to add this in.

Reply:
Thanks for clarifying age, makes sense